api/provider/provider.go

```diff
@@ -8,6 +8,7 @@ import (
 )
 
 const (
+	userNameKey  = "user_name"
```

Comment:
I would prefer not to store a new constant here unless they are common metadata information across other oauth providers.

In this context, the github user info contains a "login" key which refers to the github username while the twitter user info contains a "screen_name" key which refers to the user's twitter handle (the one that starts with @). 
I think it is better if we keep the metadata information consistent with what the github / twitter API returns to gotrue. What do you think @awalias?

Reply:
hey thanks for your feedback @kangmingtay !

both github and twitter refer to it as usernames:
https://help.twitter.com/en/managing-your-account/change-twitter-handle
https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/changing-your-github-username

and I also think other services have usernames for their users and expose them, that's why I thought I would make sense to introduce the general "user_name" variable here additionally to the already existing "full_name" variable

Reply:
@dsumer got it, yeah that makes sense! thanks for the clarification